core/src/main/java/hudson/security/FullControlOnceLoggedInAuthorizationStrategy.java

```diff
@@ -36,30 +37,52 @@
 
 /**
  * {@link AuthorizationStrategy} that grants full-control to authenticated user
- * (other than anonymous users.)
+ * and optionally read access to anonymous users
  *
  * @author Kohsuke Kawaguchi
  */
 public class FullControlOnceLoggedInAuthorizationStrategy extends AuthorizationStrategy {
+    /**
+     * Whether to allow anonymouos read access, default behavior
+     * previously was to do so
+     */
+    private boolean authenticatedReadOnly = false;
```

Comment:
Potentially misleading field name. I would read this as meaning that, if set, authenticated users would get read-only access to Jenkins. Which is of course not what you mean.

Perhaps `forbidAnonymousAccess`?